### PR TITLE
KeyFile: slim down

### PR DESCRIPTION
**Pull request by danrabbit**
Almost none of these properties are used. These constants just point to other constants. I feel like I'm taking crazy pills

### CODE REVIEW

File: src/Startup/Backend/KeyFile.vala [hunk 1]
```diff
@@ -248,18 +161,11 @@ public class Startup.Backend.KeyFile : GLib.Object {
...
-    }
-
     public Entity.AppInfo create_app_info () {
         return Entity.AppInfo () {
-            name = name,
-            comment = comment,
-            icon = icon,
+            name = keyfile_get_locale_string (KeyFileDesktop.KEY_NAME),
+            comment = keyfile_get_locale_string (KeyFileDesktop.KEY_COMMENT),
+            icon = keyfile_get_string (KeyFileDesktop.KEY_ICON),
```
Comment (tintou) [inline src/Startup/Backend/KeyFile.vala:168]: It is technically
```suggestion
            icon = keyfile_get_locale_string (KeyFileDesktop.KEY_ICON),
```

File: src/Startup/Backend/KeyFile.vala [hunk 2]
```diff
@@ -25,60 +25,34 @@
...
-        set { set_key (KEY_ICON, value); }
-    }
+    private const string FALLBACK_ICON = "application-default-icon";
+    private const string KEY_ACTIVE = "X-GNOME-Autostart-enabled";
 
     public bool active {
-        get { return get_bool_key (KEY_ACTIVE); }
-        set { set_bool_key (KEY_ACTIVE, value); }
+        get {
+            return keyfile_get_string (KEY_ACTIVE) == "true";
```
Comment (tintou) [inline src/Startup/Backend/KeyFile.vala:33]:
```suggestion
            return keyfile.get_boolean (KeyFileDesktop.GROUP, KEY_ACTIVE);
```

File: src/Startup/Backend/KeyFile.vala [hunk 3]
```diff
@@ -25,60 +25,34 @@
...
+        }
+        set {
+            var as_string = value ? "true" : "false";
+            keyfile.set_string (KeyFileDesktop.GROUP, KEY_ACTIVE, as_string);
```
Comment (tintou) [inline src/Startup/Backend/KeyFile.vala:37]:
```suggestion
            keyfile.set_boolean (KeyFileDesktop.GROUP, KEY_ACTIVE, value);
```

**Review by tintou — CHANGES_REQUESTED** [review]

**Review by jeremypw — APPROVED** [review]
No regression found so I guess this code is not needed.

**Pull request merged**